transport/src/main/java/io/netty/channel/SendDirectByteBufPool.java

```diff
@@ -0,0 +1,1189 @@
+            return buffer.nioBuffers();
+        }
+
+        @Override
+        public ByteBuffer[] nioBuffers(int index, int length) {
+            return buffer.nioBuffers(index, length);
+        }
+
+        @Override
+        public boolean hasArray() {
+            return buffer.hasArray();
+        }
+
+        @Override
+        public byte[] array() {
+            return buffer.array();
+        }
+
+        @Override
+        public int arrayOffset() {
+            return buffer.arrayOffset();
+        }
+
+        @Override
+        public boolean hasMemoryAddress() {
+            return buffer.hasMemoryAddress();
+        }
+
+        @Override
+        public long memoryAddress() {
+            return buffer.memoryAddress();
+        }
+
+        @Override
+        public String toString(Charset charset) {
+            return buffer.toString(charset);
+        }
+
+        @Override
+        public String toString(int index, int length, Charset charset) {
+            return buffer.toString(index, length, charset);
+        }
+
+        @Override
+        public int hashCode() {
+            return buffer.hashCode();
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            return buffer.equals(obj);
+        }
+
+        @Override
+        public int compareTo(ByteBuf buffer) {
+            return this.buffer.compareTo(buffer);
+        }
+
+        @Override
+        public String toString() {
+            return buffer.toString();
+        }
+
+        @Override
+        public ByteBuf retain(int increment) {
+            return retain0(checkPositive(increment, "increment"));
+        }
+
+        @Override
+        public ByteBuf retain() {
+            return retain0(1);
+        }
+
+        private ByteBuf retain0(int increment) {
+            assertThread();
+            if (refCnt <= 0) {
+                throw new IllegalReferenceCountException();
+            }
+            refCnt += increment;
+            return this;
+        }
+
+        @Override
+        public ByteBuf touch() {
+            return buffer.touch();
+        }
+
+        @Override
+        public ByteBuf touch(Object hint) {
+            return buffer.touch(hint);
+        }
+
+        @Override
+        public int refCnt() {
+            assertThread();
+            return refCnt;
+        }
+
+        @Override
+        public boolean release() {
+            return release0(1);
+        }
+
+        @Override
+        public boolean release(int decrement) {
+            return release0(checkPositive(decrement, "decrement"));
+        }
+
+        private boolean release0(int decrement) {
+            assertThread();
+            if (refCnt < decrement) {
+                throw new IllegalReferenceCountException();
+            }
+            refCnt -= decrement;
+
+            final Preallocation parent = this.parent;
+            if (refCnt == 0) {
+                if (-- parent.refCnt == 0) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1066)